CargoMonitor/CargoMonitor.cs

```diff
@@ -583,6 +590,134 @@ public void _handleLimpetLaunchedEvent()
+                    {
+                        cargo = GetCargoWithMissionId(@event.missionid ?? 0);
+                        if (cargo != null)
+                        {
+                            // Cargo instantiated by either 'Mission accepted' event, previous 'WingUpdate' or 'Collect' updates 
+                            haulageAmount = cargo.haulageamounts.FirstOrDefault(ha => ha.id == @event.missionid);
+
+                            //Update commodity definition if intantiated by previous 'WingUpdate' update
+                            if (cargo.commodityDef.edname == "Unknown")
+                            {
+                                cargo.commodityDef = @event.commodityDefinition;
+                            }
+                        }
+                        else
+                        {
+                            // Cargo instantiated by previous 'Market buy' event
+                            cargo = GetCargoWithEDName(@event.commodityDefinition.edname);
+
+                            string type = @event.collected > 0 ? "MISSION_DeliveryWing" : "MISSION_CollectWing";
+                            haulageAmount = new HaulageAmount(@event.missionid ?? 0, type, amountRemaining, DateTime.MaxValue, true);
+                            cargo.haulageamounts.Add(haulageAmount);
+                        }
+
+                        if (haulageAmount.type == "deliverywing")
+                        {
+                            cargo.haulage -= @event.amount ?? 0;
+                        }
+                        else
+                        {
+                            cargo.owned -= @event.amount ?? 0;
+                        }
+                        cargo.CalculateNeed();
+                        haulageAmount.collected = @event.collected;
+                        haulageAmount.delivered = @event.delivered;
+
+                        // Check for shared mission completion
+                        if (haulageAmount.shared && amountRemaining == 0)
+                        {
+                            cargo.haulageamounts.Remove(haulageAmount);
+                            RemoveCargo(cargo);
+                        }
+                    }
+                    break;
+                case "WingUpdate":
+                    {
+                        cargo = GetCargoWithMissionId(@event.missionid ?? 0);
+                        if (cargo != null)
+                        {
+                            // Cargo instantiated by either 'Mission accepted' event, previous 'WingUpdate' or 'Collect' updates
+                            haulageAmount = cargo.haulageamounts.FirstOrDefault(ha => ha.id == @event.missionid);
+                            haulageAmount.amount = amountRemaining;
+                        }
+                        else
+                        {
+                            // First exposure to new cargo, use 'Unknown' as placeholder
+                            cargo = new Cargo("Unknown", 0);
```

Comment:
Update this if we elect to use a more descriptive name...

Reply:
'Unknown' is just the EDName.